frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -52,7 +58,61 @@ import {
   getPort,
 } from '../kms-config/utils';
 import './ocs-storage-class-form.scss';
-import { GUARDED_FEATURES } from '../../features';
+
+export const CephFsNameComponent: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+
+  const isExternal = useFlag(OCS_INDEPENDENT_FLAG);
+  const scName = `${isExternal ? OCS_EXTERNAL_CR_NAME : OCS_INTERNAL_CR_NAME}-cephfs`;
+  const [sc, scLoaded, scLoadError] = useK8sGet<StorageClassResourceKind>(
+    StorageClassModel,
+    scName,
+  );
+  const [fileSystemName, setFileSystemName] = React.useState<string>('');
+
+  const onFsNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const name = e.currentTarget.value;
+    setFileSystemName(name);
+    onParamChange(name);
+  };
+
+  React.useEffect(() => {
+    if (scLoaded && !scLoadError) {
+      const fsName = sc?.parameters?.fsName;
+      if (fsName) {
+        setFileSystemName(fsName);
+        onParamChange(fsName);
+      }
+    }
+    /** onParamChange will trigger a state update in parent component 
+    which will cause this hook to run again causing maximum depth exceeded error */
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sc, scLoaded, scLoadError, setFileSystemName]);
```

Comment:
Have you tried using `React.memo()`. Try using it with `_.isEqual`. Maybe that would fix the maximum depth exceeded error.